flytekit/remote/remote.py

```diff
@@ -992,20 +1008,48 @@ def _get_output_literal_map(self, execution_data: ExecutionDataResponse) -> lite
                 )
         return literal_models.LiteralMap({})
 
-    def _get_node_execution_interface(self, node_execution: FlyteNodeExecution) -> TypedInterface:
+    def _get_node_execution_interface(
+        self, node_execution: FlyteNodeExecution, entity_definition: typing.Union[FlyteWorkflow, FlyteTask]
+    ) -> TypedInterface:
         """Return the interface of the task or subworkflow associated with this node execution."""
-        if not node_execution.metadata.is_parent_node:
+        if isinstance(entity_definition, FlyteTask):
+            # A single task execution consists of a Flyte workflow with single node whose interface matches that of
+            # the underlying task
+            return entity_definition.interface
+
+        for node in entity_definition.flyte_nodes:
+            if node.id == node_execution.id.node_id:
+                if node.task_node is not None:
+                    return node.task_node.flyte_task.interface
+                elif node.workflow_node is not None and node.workflow_node.sub_workflow_ref is not None:
+                    # Fetch the workflow and use its interface
+                    sub_workflow_ref = node.workflow_node.sub_workflow_ref
+                    workflow = self.fetch_workflow(
+                        sub_workflow_ref.project,
+                        sub_workflow_ref.domain,
+                        sub_workflow_ref.name,
+                        sub_workflow_ref.version,
+                    )
+                    return workflow.interface
+                elif node.workflow_node is not None and node.workflow_node.launchplan_ref is not None:
```

Comment:
add test for this launch plans?

Reply:
@cosmicBboy 
I tried adding an integration test but Yee informs me it's broken, same with subworkflows, see

```
@classmethod
    def promote_from_model(
        cls,
        base_model: _workflow_model.WorkflowNode,
        sub_workflows: Dict[_identifier.Identifier, _workflow_model.WorkflowTemplate],
        tasks: Dict[_identifier.Identifier, _task_model.TaskTemplate],
    ) -> "FlyteWorkflowNode":
        from flytekit.remote import launch_plan as _launch_plan
        from flytekit.remote import workflow as _workflow
    
        fetch_args = (
            base_model.reference.project,
            base_model.reference.domain,
            base_model.reference.name,
            base_model.reference.version,
        )
    
        if base_model.launchplan_ref is not None:
>           return cls(flyte_launch_plan=_launch_plan.FlyteLaunchPlan.fetch(*fetch_args))
E           AttributeError: type object 'FlyteLaunchPlan' has no attribute 'fetch'
```

Reply:
@cosmicBboy can I remove the test and re-add once this is fixed?

Reply:
yes! I totally missed this method, those `fetch`es aren't meant to be there.

if you want to keep that test code there we can decorate it with `@pytest.mark.skip(reason="some reason")` and delete it with the fix PR

Reply:
thanks @cosmicBboy updated

Reply:
adding this to https://github.com/flyteorg/flyte/issues/1353